utils.js

```diff
@@ -10540,3 +10541,52 @@ export function parseMakeDFile(dfile) {
   pkgFilesMap[pkgName] = Array.from(filesList);
   return pkgFilesMap;
 }
+
+/**
+ * Function to validate an externalReference URL for conforming to the JSON schema or bomLink
+ * https://github.com/CycloneDX/cyclonedx-core-java/blob/75575318b268dda9e2a290761d7db11b4f414255/src/main/resources/bom-1.5.schema.json#L1140
+ * https://datatracker.ietf.org/doc/html/rfc3987#section-2.2
+ * https://cyclonedx.org/capabilities/bomlink/
+ *
+ * @param {String} url URL to validate
+ *
+ * @returns {Boolean} Flag indicating whether the supplied URL is valid or not
+ *
+ */
+export function isValidIriReference(url) {
```

Comment:
Wondering if this can be replaced with a simple [url](https://nodejs.org/api/url.html) parse and protocol checks?

Reply:
@timmyteo was doing a try/catch on `new URL()`, but it was allowing urls that are rejected by DependencyTrack. According to my tests, this is the best option to be compatible with DT (I also tried to validate using https://github.com/luzlab/ajv-formats-draft2019/blob/master/formats/iri-reference.js, but it was too permissive too).
Anyway, it needs more testing.